chemprop/cli/predict.py

```diff
@@ -282,12 +277,9 @@ def make_prediction_for_models(
         individual_preds.append(preds)
 
     average_preds = torch.mean(torch.stack(individual_preds).float(), dim=0)
-    if args.target_columns is not None:
-        assert (
-            len(args.target_columns) == model.n_tasks
-        ), "Number of target columns must match the number of tasks."
-        target_columns = args.target_columns
-    else:
+
+    target_columns = load_target_columns(model_paths[0])
+    if target_columns is None:
```

Comment:
```suggestion
    output_columns = load_output_columns(model_paths[0])
    if output_columns is None:
```

Reply:
Can we just condense `load_model` and `load_target_columns` into one function? If we're saving them in the same structured file, I don't see the benefit to reading a file twice.

Reply:
Yeah, we can condense them into one function. However, it will still require reading the file twice because the model  file is read within the attribute `load_from_file`, but not in the load_model

```
def load_model(path: PathLike, multicomponent: bool) -> MPNN:
    if multicomponent:
        model = MulticomponentMPNN.load_from_file(path)
    else:
        model = MPNN.load_from_file(path)

    return model
```

Reply:
How about always returning `output_columns` from the `load_from_file` function?

Reply:
Since `load_from_file` is the attribute of the MPNN class, I think users would only expect it to have one output and returns the loaded model, but not the additional `output_columns` though. I think the current way of `load_from_file` is also nice because you can use it interchangeably with `load_from_checkpoint`.